STYLE_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+# Front-end style guide
+
+This document outlines recommended code style, quirks, and best practices for front-end development on Kedro-Viz.
+
+<!-- @TODO table of contents -->
+
+## Code style and linting
+
+We use [Prettier](https://prettier.io/), [ESLint](https://eslint.org/), and [stylelint](https://stylelint.io/) to lint our SCSS and JavaScript. For the most part, we use the default config - you can find specific details in the `.prettierrc`, `.eslintrc.json`, and `.stylelintrc` config files. We recommend that you install the Prettier, ESLint, and stylelint plugins for your text-editor, and enable automatic formatting on save.
+
+Other than that, we mostly recommend following the [AirBnB JavaScript Style Guide](https://github.com/airbnb/javascript).
+
+##  Browser and device support
+
+We aim to support recent versions of major modern browsers - i.e. Chome, Firefox, Edge, Safari, and Opera, on both MacOS and Windows. Keeping progressive enhancement in mind, we aim for decent mobile support where possible, i.e. nothing should be broken on mobile, but it's okay for mobile/older browsers to have a slightly degraded experience.
```

Comment:
Do you have a preferred way of testing across the range? Maybe this is obvious, or do you use browserstack or something similar? Maybe worth mentioning.

Reply:
We honestly don't, which is a wider problem and something I should discuss with the QB Labs front-end team. Thanks for noting it!